sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/client/ComponentCall.scala

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.javasdk.impl.client
```

Comment:
I'm not sure about packaging here and for `MethodRefResolver`

Reply:
`MethodRefResolver` should stay in impl as it is not directly exposed to end users, if I'm not mistaking.

On the other hand, `ComponentCall` to `ComponentCall22` must be public and we need to offer guarantees. Also good if we make them `final`.

Reply:
ComponentCall still need to move to public package. It's user land api.

Reply:
final added

Reply:
that's a bit problematic, because it's a scala code, or you mean to just change the package name?

Reply:
do you think I should rewrite it in Java?

Reply:
ok, moved to a proper package